xwiki-platform-core/xwiki-platform-attachment/xwiki-platform-attachment-ui/src/main/resources/XWiki/AttachmentSelector.xml

```diff
@@ -434,6 +463,68 @@ $xwiki.jsx.use($attachmentPickerDocName)
     </property>
     <property>
       <code>var XWiki = (function(XWiki) {
+  function uploadTemporaryAttachment() {
+    // Require jquery locally until we are able to fully migrate this code away from prototype.
+    const form = this.property.up('form');
+    require(['jquery'], function ($) {
+      const data = new FormData();
+      const uploadedFile = $('#attachfile')[0].files[0];
+      const filenameCheckbox = $("#uploadAttachment input[name='filename']");
+  
+      // TODO: Fix replace currently selected checkbox (see XWIKI-20181).
+      data.append('upload', uploadedFile);
+      const notification = new XWiki.widgets.Notification(
+        "$services.localization.render('xe.attachmentSelector.upload.inProgress')", 'inprogress');
+      const params = {
+        'form_token': $(form).find('[name="form_token"]').val(),
+        'sheet': 'CKEditor.FileUploader',
+        'outputSyntax': 'plain'
+      };
+  
+      // Update the name of the file with the name of the currently selected attachment if required.
+      if (filenameCheckbox.prop('checked')) {
+        params['filename'] = filenameCheckbox.val()
+      }
+  
+      $.ajax({
+        'url': XWiki.currentDocument.getURL('get', new URLSearchParams(params).toString()),
+        method: 'POST',
+        data: data,
+        processData: false,
+        contentType: false, // Sets the 'multipart/form-data' automatically
+        headers: {
+          'X-XWiki-Temporary-Attachment-Support': true
+        }
+      }).done(function (response) {
+        #set ($successMessage = $escapetool.javascript($services.localization.render('attachment.attachmentSelector.temporaryUpload.success')))
+        notification.replace(new XWiki.widgets.Notification("$successMessage", 'done'));
+        // Add a field with the name of the uploaded file so that it is moved from the temporary
+        // attachments to the persisded ones on save.
+        $(form).append($('&lt;input/&gt;')
+          .prop('type', 'hidden')
+          .prop('name', 'uploadedFiles')
+          .prop('value', response.fileName))
+        $(form).find('input[type="hidden"].property-reference').prop('value', response.fileName);
+        this.updateAttachment(response.fileName, response.url);
+        this.dialog.closeDialog();
+      }.bind(this)).fail(function  () {
+        #set ($errorMessage = $escapetool.javascript($services.localization.render('attachment.attachmentSelector.temporaryUpload.failure')))notification.replace(new XWiki.widgets.Notification("$errorMessage", 'error'));
+      });
+    }.bind(this));
+  }
+  
+  function directUploadAttachment(uploadForm) {
+    // FIXME This fails in HTML5, will deal with it later:
```

Comment:
what does it mean exactly? UC not supported with html5?

Reply:
This is an old comment I moved when refactoring. Maybe @mflorea knows if this is still accurate.